File: apps/sdkwork-appstore-mini-program/packages/sdkwork-appstore-mp-charts/src/services/chartsService.ts
import type { AppstoreAppClient } from "@sdkwork/appstore-mp-core";

import type { ChartsPageResult } from "../types/chartsModels.js";

/**
 * Charts service.
 *
 * The generated App Store app SDK client is injected by the root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export interface ChartsService {
  readonly capability: "charts";
}

export function createChartsService(client: AppstoreAppClient): ChartsService {
  void client;
  return { capability: "charts" };
}

export function emptyChartsResult(): ChartsPageResult<never> {
  return { items: [] };
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_charts/lib/src/services/charts_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/charts_models.dart';

/// Charts service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class ChartsService {
  const ChartsService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'charts';

  ChartsPageResult<Never> empty() =>
      const ChartsPageResult<Never>(items: <Never>[]);
}
